File: php/concepts/operators.php
<?php

echo "\n=== Operators ===\n";
$a = 10;
$b = 3;

// Arithmetic Operators
echo "Arithmetic Operators\n";
echo "Addition: " . ($a + $b) . PHP_EOL;
echo "Subtraction: " . ($a - $b) . PHP_EOL;
echo "Multiplication: " . ($a * $b) . PHP_EOL;
// The division operator `/` returns a float value unless the two operands
// are integers and the numbers are evenly divisible.
echo "Division: " . ($a / $b) . PHP_EOL;

// Comparison Operators
echo "\nComparison Operators\n";
echo "Is a > b? " . ($a > $b) . PHP_EOL; // Prints 1 for true
// Strict equality `===` checks for value AND type. This is best practice.
echo "Is a === b? " . ($a === $b) . PHP_EOL; // Prints empty string for false

File: rust/concepts/operators.rs
pub fn demonstrate_operators() {
    println!("\n=== Operators ===");
    let a = 10; // Inferred as i32
    let b = 3;  // Inferred as i32

    // Arithmetic Operators
    println!("Arithmetic Operators");
    println!("Addition: {}", a + b);
    println!("Subtraction: {}", a - b);
    println!("Multiplication: {}", a * b);

    // Integer division truncates the result.
    println!("Integer Division: {}", a / b);

    // To get a float result, we must cast the types.
    println!("Float Division: {}", a as f64 / b as f64);

    // Comparison Operators
    println!("\nComparison Operators");
    println!("Is a > b? {}", a > b);
    println!("Is a == b? {}", a == b);
}
